mantidproject/mantid
# Add SetCrystalLocation

New algorithm `SetCrystalLocation` to set the sample position of an events workspace.  This provides a nice, simple interface to adjust sample position when recreating MD workspaces with samples not at the origin.

No tests have been added since it changes the sample position the same way that `OptimizeCrystalPlacement` does, which already has tests.  

**To test:**
```python
events = Load('TOPAZ_3007_bank_37_20_sec.nxs')
sample = mtd['events'].getInstrument().getSample()
print('Sample position before SetCrystalLocation: {}'.format(sample.getPos()))
SetCrystalLocation(InputWorkspace=events, OutputWorkspace=events, NewX=0.1, NewY=0.1, NewZ=0.1)
print('Sample position after SetCrystalLocation: {}'.format(sample.getPos()))
```
You can also verify that MDWorkspaces created from these are different.  Note that if you test that, you must delete the preprocessed detectors workspace or name a new preprocessing workspace when calling `ConvertToMD`.

Fixes #24329 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
